src/Apps/Order/Components/AddressForm.tsx

```diff
@@ -85,7 +86,9 @@ export class AddressForm extends React.Component<
   }
 
   phoneNumberInputDescription = (): string | null => {
-    if (this.props.billing && this.props.showPhoneNumberInput) {
+    if (this.props.phoneNumberOnly) {
```

Comment:
Thoughts on adding a separate component for this phone number input instead of trying to reuse `AddressForm`? There's a lot of conditional logic added, seems like it would be simpler/easier to reason about if it were separate.

Reply:
I did it that way at first, but I found myself duplicating so much that this seemed like a better option

Reply:
What was duplicated? It seems like you can keep the input fields the same but create a version of this component that is pared down to just the phone number input. Would that work?

Since this component is used across the site where we need people to add shipping/billing addresses for credit cards (auctions, user settings payments, orders), it feels odd to extend it for this purpose.

We should probably (not as part of this PR 😄) re-name this component and extract it from the `Order` sub-app.

Reply:
Like the handlers and stuff. It just felt like a lot of duplication when I started doing it that way

Reply:
I talked it over with Ashkan and he thought this way was fine but can do it that way if you think its better